Dfc.CourseDirectory.Services.Tests/BulkUploadService/Apprenticeship/LocationVariationFacts.cs

```diff
@@ -0,0 +1,558 @@
+                    venue = DefaultVenue,
+                    region = "South West",
+                    subRegion = "Devon;Junk",
+                    expectedError = "Validation error on row 2. Field SUB_REGION  contains invalid SubRegion names",
+                },
+                new LocationVariationFact
+                {
+                    deliveryMethod = "employer",
+                    deliveryMode = "",
+                    acrossEngland = "",
+                    nationalDelivery = "no",
+                    venue = DefaultVenue,
+                    region = "South West",
+                    subRegion = "Junk",
+                    expectedError = "Validation error on row 2. Field SUB_REGION  contains invalid SubRegion names",
+                },
+                new LocationVariationFact
+                {
+                    deliveryMethod = "employer",
+                    deliveryMode = "",
+                    acrossEngland = "",
+                    nationalDelivery = "no",
+                    venue = DefaultVenue,
+                    region = "south west",
+                    subRegion = "devon",
+                    expectedError = NoError,
+                    expectedOutputDeliveryMode = null,
+                },
+                new LocationVariationFact
+                {
+                    deliveryMethod = "employer",
+                    deliveryMode = "",
+                    acrossEngland = "",
+                    nationalDelivery = "yes",
+                    venue = "",
+                    region = "",
+                    subRegion = "",
+                    expectedError = NoError,
+                    expectedOutputDeliveryMode = null,
+                },
+                new LocationVariationFact
+                {
+                    deliveryMethod = "employer",
+                    deliveryMode = "",
+                    acrossEngland = "",
+                    nationalDelivery = "yes",
+                    venue = DefaultVenue,
+                    region = "",
+                    subRegion = "",
+                    expectedError = NoError,
+                    expectedOutputDeliveryMode = null,
+                },
+                new LocationVariationFact
+                {
+                    deliveryMethod = "junk",
+                    deliveryMode = "",
+                    acrossEngland = "",
+                    nationalDelivery = "",
+                    venue = "",
+                    region = "",
+                    subRegion = "",
+                    expectedError = "Validation error on row 2. Field DELIVERY_METHOD is invalid.",
+                },
+            };
+            foreach (var fact in facts)
+            {
+                Add(fact);
+            }
+
+            /* Test analysis code; uncomment this to generate a csv of all the covered variations which can then be used to check that all desired variations are covered and to facilitate discussions.
```

Comment:
If we need this here the `WriteAllText` should work cross-plat (or at least on Windows). `Path.GetTempPath` may be helpful.

Reply:
Done, thanks for the push to improve it.